### DISCUSSION TITLE
Using the Template tp object outside of Templater scripts? like in the Console or CustomJS?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [SilentVoid13/Templater#1247] [Help]
For instance, I want to use `tp.file.create_new()` inside either the devtools console or in a CustomJS script, even if I could do what those templater functions with the Obsidian API directly.

**Comment by Zachatoo** [comment]
You can read through the Templater source code to see what Obsidian API methods Templater is calling, and do something similar.

I'd recommend against calling Templater directly since Templater is stateful, you'll need to perform setup and teardown before and after your script respectively for everything to work properly.

If you call Templater commands using `app.commands.executeCommandById()`, then Templater can handle that for you, but it's usually not very useful in scripts unless you create commands for specific templates (see hotkeys in Templater settings).
